x-pack/plugins/fleet/server/services/api_keys/enrollment_api_key.ts

```diff
@@ -182,19 +176,30 @@ export async function generateEnrollmentAPIKey(
   }
 
   const name = providedKeyName ? `${providedKeyName} (${id})` : id;
-  const key = await createAPIKey(soClient, name, {
-    // Useless role to avoid to have the privilege of the user that created the key
-    'fleet-apikey-enroll': {
-      cluster: [],
-      applications: [
-        {
-          application: '.fleet',
-          privileges: ['no-privileges'],
-          resources: ['*'],
+
+  const { body: key } = await esClient.security
+    .createApiKey({
```

Comment:
Should we add some metadata to these enrollment keys?

Maybe it would be nice to have this defined as a constant outside which makes it easier to document and reference.

Reply:
yes we could add some metadata, what do you have in mind here?

Reply:
In https://github.com/elastic/fleet-server/pull/195 we have:

```
"metadata" : {
        "agent_id" : "e4dede19-759e-45d5-b08f-0e78dec888e5",
        "managed_by" : "fleet-server",
        "managed" : true,
        "type" : "output"
      }
```

We could do something similar here:

```
"metadata" : {
        "managed_by" : "fleet",
        "managed" : true,
        "type" : "enroll" (or similar)
        "policy_id": "1234"
      }
```

Reply:
Great add this here https://github.com/elastic/kibana/pull/96464/commits/9833563d2ea3d907f4908e71bfdec0a298a4e9f9